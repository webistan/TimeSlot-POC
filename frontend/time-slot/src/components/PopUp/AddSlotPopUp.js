import React, { useContext, useState } from "react";

import { MyContext } from "../SlotCard/SlotCard";
import _ from "lodash";
import { addSlotInObject } from "../../redux/action/SlotAction";
import { connect } from "react-redux";
import moment from "moment";

//***************************** AddSlotPopUp Method **********************************//
const AddSlotPopUp = (props) => {
  //const { addSlotInObject } = props;
  const {
    addSlotsObj,
    day,
    open,
    onClose,
    slotsObj,
    slotList,
    slotCopyDay,
    copySolts,
    removeSlotsObj,
  } = useContext(MyContext);

  const [slotTime, setSlotTime] = useState("");
  const [slotNumber, setSlotNumber] = useState("");

  // console.log("slotList are::: ", slotsObj);
  // console.log('slotCopyDay are:: ',slotCopyDay)

  
  //***************************** Close PopUp Method **********************************//
  const onClosePopUp = () => {
    let json = {
      popUpClose: true,
      addData: false,
      selectedDay: day,
    };
    onClose(json);
  };

  //***************************** AddSlot Button Click Method **********************************//
  const addSlot = () => {
    let json = {
      popUpClose: true,
      addData: true,
      obj: slotsObj,
      selectedDay: day,
    };
    onClose(json);
  };

  const onChangeSlotTime = (e) => {
    let timetxt = `${e.target.value}:00`;
    setSlotTime(timetxt);
  };
  const onChangeSlotNumber = (e) => {
    setSlotNumber(e.target.value);
  };

  //***************************** Method Of Adding Slot In PoPUp  **********************************//
  const addSlotInCard = (e) => {
    if (!_.isEmpty(slotsObj)) {
      let arr = slotsObj[day];
      console.log("arr5", arr);
      arr.push({
        start_time: slotTime,
        allocated_slot: slotNumber,
        allocated_day: day,
      });
      let json = {
        [day]: arr,
      };
      // console.log("json2",json)
      addSlotsObj(json);
    } else {
      let arr = [];
      // console.log("arr5", arr);
      arr.push({
        start_time: slotTime,
        allocated_slot: slotNumber,
        allocated_day: day,
      });
      let json = {
        [day]: arr,
      };
      console.log("json2", json);
      addSlotsObj(json);
    }
    setSlotNumber("");
    setSlotTime("");
  };

  //***************************** Method Of Cancel Slots  In PoPUp  **********************************//
  const cancelTimeSlot = (item, idx) => {
    let json = {
      item: item,
      index: idx,
    };
    removeSlotsObj(json);
  };

  //***************************** Copy Schedule Method **********************************//
  const setCopyDay = (copyDay) => {
    copySolts(copyDay);
  };

  // console.log('slotsObj are:: ',slotsObj)

  return (
    <div className="modal" data-testid="slot-popup">
      <div className="wt-card">
        <div className="wt-card-head">
          <div className="card-head-left">Timeslot</div>
          <div className="card-head-right">
            <button className="close" onClick={onClosePopUp}>
              X
            </button>
          </div>
        </div>
        <div>
          <div className="input-group">
            <input
              type="time"
              className="w-130"
              defaultValue={slotTime}
              onChange={onChangeSlotTime}
            />
            <input
              type="text"
              className="w-90"
              defaultValue={slotNumber}
              onChange={onChangeSlotNumber}
            />
            <button
              className={"add-btn "}
              disabled={slotNumber === "" || slotTime === "" ? true : false}
              onClick={addSlotInCard}
            >
              <i
                className="fa-sharp fa-solid fa-plus"
                style={{ color: "#ff4eb2" }}
              />
            </button>
          </div>
          <div className="tag-outer">
            {slotsObj &&
              !_.isEmpty(slotsObj) &&
              slotsObj[day].map((item, idx) => {
                return (
                  <div className="tag" key={idx}>
                    {moment(item.start_time, "HH:mm:ss").format("hh:mm A")}/
                    <span>{item.allocated_slot}Slots</span>
                    <div
                      className="close-tag"
                      onClick={() => cancelTimeSlot(item, idx)}
                    >
                      X
                    </div>
                  </div>
                );
              })}
          </div>
          <div className="bottom-block" >
            <h4>Copy Schedule</h4>
            <div className="days">
              {slotCopyDay &&
                slotCopyDay.map((item, idx) => {
                  return (
                    <>
                      <span className="copy-day" data-testid="copy-day" key={idx}>
                        <button
                          className={
                            "btn1 " + (item.isSelected ? "btnhover " : "")
                          }
                          onClick={(e) => setCopyDay(item.day)}
                        >
                          {item.showDay}
                        </button>
                      </span>
                    </>
                  );
                })}
            </div>
          </div>
          <button
            className={
              "modal-btn " + (!_.isEmpty(slotsObj) ? "button2" : "button-dis")
            }
            disabled={_.isEmpty(slotsObj) ? true : false}
            onClick={addSlot}
          >
            + Add Slot
          </button>
        </div>
      </div>
    </div>
  );
}
//***************************** MapStateToProps Method **********************************//
// const mapStateToProps = (state) => ({});

// //***************************** MapDispatchToProps Method **********************************//
// const mapDispatchToProps = {
//   //addSlotInObject,
// };

//export default connect(mapStateToProps, mapDispatchToProps)(AddSlotPopUp);

export default AddSlotPopUp