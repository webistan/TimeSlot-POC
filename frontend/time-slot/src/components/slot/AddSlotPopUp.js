import React, { useContext } from 'react'

import { MyContext } from "./SlotCard"

function AddSlotPopUp(props) {
    // console.log("pro123",props)
    const { open, onClose } = useContext(MyContext);

    const onClosePopUp = () => {
      let json = {
        popUpClose: true,
        addData: false
      }
      onClose(json)
    }

    const addSlot = () => {
      let json = {
        popUpClose: true,
        addData: true
      }
      onClose(json)
    }
  return (
    <div className="modal">
        <div className="wt-card">
          <div className="wt-card-head">
            <div className="card-head-left">Timeslot</div>
            <div className="card-head-right">
              <button className="close" onClick={onClosePopUp}>X</button>
            </div>
          </div>
          <div>
            <div className="input-group">
              <input type="time" className="w-130" />
              <input type="text" className="w-90" />
              <span className="delete-btn"><i className="fa-sharp fa-solid fa-plus" style={{color: '#ff4eb2'}} /></span>
            </div>
            <div className="tag-outer">
              <div className="tag">
                09:30AM/<span>10Slots</span>
                <div className="close-tag">X</div>
              </div>
              <div className="tag">
                09:30AM/<span>10Slots</span>
                <div className="close-tag">X</div>
              </div>
              <div className="tag">
                09:30AM/<span>10Slots</span>
                <div className="close-tag">X</div>
              </div>
              <div className="tag">
                09:30AM/<span>10Slots</span>
                <div className="close-tag">X</div>
              </div>
            </div>
            <div className="bottom-block">
              <h4>Copy Schedule</h4>
              <div className="days">
                <button className="btn1">S</button>
                <button className="btn1 btnhover">M</button>
                <button className="btn1">T</button>
                <button className="btn1">W</button>
                <button className="btn1 btnhover">Th</button>
                <button className="btn1">F</button>
                <button className="btn1">S</button>
              </div>
            </div>
            <button className="button2 modal-btn" onClick={addSlot}>+ Add Slot</button>
          </div>
        </div>
      </div>
  )
}

export default AddSlotPopUp