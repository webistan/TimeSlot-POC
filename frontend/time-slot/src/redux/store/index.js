import { applyMiddleware, createStore } from 'redux'

import { createLogger } from 'redux-logger'
import createSagaMiddleware from 'redux-saga'
import rootReducer from '../../redux/reducer'
import sagas from '../../redux/saga'

// import sagas from './redux/saga';
// import rootReducer from './redux/reducer';

const loggerMiddleware = createLogger()

const sagaMiddleware = createSagaMiddleware()

const store = createStore(rootReducer, applyMiddleware(sagaMiddleware,loggerMiddleware))

sagaMiddleware.run(sagas)

export default store

