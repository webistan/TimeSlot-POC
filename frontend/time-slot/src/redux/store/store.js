import { configureStore } from "@reduxjs/toolkit";
import slot from "../reducer/index";

export const createStore = () =>
  configureStore({
    reducer: {
      slotReducer: slot,
    },
  });
