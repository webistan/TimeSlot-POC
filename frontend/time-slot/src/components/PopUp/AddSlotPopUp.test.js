// Importing the jest testing library
import '@testing-library/jest-dom'

import { cleanup, render, screen } from "@testing-library/react";

import AddSlotPopUp from "./AddSlotPopUp";
import { Component } from 'react';
import { Provider } from 'react-redux';
import {createStore} from "../../redux/store/store";
import renderer from 'react-test-renderer';

// afterEach function runs after each test suite is executed
afterEach(() => {
    cleanup(); // Resets the DOM after each test suite
})
  
describe("AddSlotPopUp Component" ,  () => {

    test('Should check Slot Card Pop Up be exist in Component', () => {
      const { getByTestId } = render(
        <Provider store={createStore()}>
          <AddSlotPopUp />
        </Provider>
      );
      const SlotPopUp = getByTestId("slot-popup");
      
      expect(SlotPopUp).toBeInTheDocument(); 
    });

    test('Should check Slot copy day div not exist in Component', () => {
      const slotCopyDay = [
        {
          day: "Sunday",
          showDay: "S",
          isSelected: false,
          copyDay: false,
        },
        {
          day: "Monday",
          showDay: "M",
          isSelected: false,
          copyDay: false,
        },
        {
          day: "Tuesday",
          showDay: "T",
          isSelected: false,
          copyDay: false,
        },
        {
          day: "Wednesday",
          showDay: "W",
          isSelected: false,
          copyDay: false,
        },
        {
          day: "Thursday",
          showDay: "Th",
          isSelected: false,
          copyDay: false,
        },
        {
          day: "Friday",
          showDay: "F",
          isSelected: false,
          copyDay: false,
        },
        {
          day: "Saturday",
          showDay: "Sat",
          isSelected: false,
          copyDay: false,
        },
      ]
      render(<AddSlotPopUp slotCopyDay={slotCopyDay}/>);
      const SlotPopUpDomTree = renderer.create(<AddSlotPopUp />).toJSON();

      expect(screen.queryByTestId('copy-day')).not.toBeInTheDocument();
      expect(SlotPopUpDomTree).toMatchSnapshot();
  });



})

