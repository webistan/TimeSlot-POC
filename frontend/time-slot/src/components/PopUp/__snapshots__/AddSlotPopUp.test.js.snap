// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddSlotPopUp Component Should check Slot copy day div not exist in Component 1`] = `
<div
  className="modal"
  data-testid="slot-popup"
>
  <div
    className="wt-card"
  >
    <div
      className="wt-card-head"
    >
      <div
        className="card-head-left"
      >
        Timeslot
      </div>
      <div
        className="card-head-right"
      >
        <button
          className="close"
          onClick={[Function]}
        >
          X
        </button>
      </div>
    </div>
    <div>
      <div
        className="input-group"
      >
        <input
          className="w-130"
          defaultValue=""
          onChange={[Function]}
          type="time"
        />
        <input
          className="w-90"
          defaultValue=""
          onChange={[Function]}
          type="text"
        />
        <button
          className="add-btn "
          disabled={true}
          onClick={[Function]}
        >
          <i
            className="fa-sharp fa-solid fa-plus"
            style={
              Object {
                "color": "#ff4eb2",
              }
            }
          />
        </button>
      </div>
      <div
        className="tag-outer"
      />
      <div
        className="bottom-block"
      >
        <h4>
          Copy Schedule
        </h4>
        <div
          className="days"
        />
      </div>
      <button
        className="modal-btn button-dis"
        disabled={true}
        onClick={[Function]}
      >
        + Add Slot
      </button>
    </div>
  </div>
</div>
`;
